lib/builders/texify.js

```diff
@@ -30,23 +30,38 @@ export default class TexifyBuilder extends Builder {
     const args = [
       '--batch',
       '--pdf',
-      '--tex-option=--synctex=1',
+      '--tex-option="--synctex=1"',
+      '--tex-option="--interaction=nonstopmode"',
       // Set logfile max line length.
-      '--tex-option=--max-print-line=1000'
+      '--tex-option="--max-print-line=1000"'
     ]
 
     const enableShellEscape = atom.config.get('latex.enableShellEscape')
+    const engineFromMagic = this.getLatexEngineFromMagic(filePath)
+    const customEngine = atom.config.get('latex.customEngine')
+    const engine = atom.config.get('latex.engine')
+
     if (enableShellEscape) {
       args.push('--tex-option=--enable-write18')
     }
 
-    const customEngine = atom.config.get('latex.customEngine')
-    const engine = atom.config.get('latex.engine')
-    if (customEngine || engine !== 'pdflatex') {
-      atom.notifications.addWarning(
-        'Engine customization is not supported by texify, ' +
-        'so this functionality is disabled when using the texify builder.'
-      )
+    if (engineFromMagic) {
+      args.push(`--engine="${engineFromMagic}"`)
+    } else if (customEngine) {
+      args.push(`--engine="${customEngine}"`)
+    } else if (engine && engine === 'xelatex') {
+      args.push(`--engine=xetex`)
+    } else if (engine && engine === 'lualatex') {
+      args.push(`--engine=luatex`)
+    }
+
+    if (engine) {
```

Comment:
This entire `if (engine)` block can be removed since it's a duplication of the `if` chain above.